tests/runner/src/fluvio-integration-derive/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+
+use fluvio_test_util::test_meta::FluvioTestRequirements;
+use syn::{AttributeArgs, Ident, ItemFn, parse_macro_input};
+use quote::quote;
+
+/// This macro will allow a test writer to override
+/// minimum Fluvio cluster requirements for a test
+///
+/// Supported keys: `min_spu` (default 1)
+///
+/// #[fluvio_test(min_spu = 2)]
+/// pub async fn run(_client: Fluvio, option: TestOption) {
+///     println!("Test")
+/// }
+///
+/// Will expand into
+///
+/// pub async fn run(_client: Fluvio, option: TestOption) {
+///    use fluvio_test_util::test_meta::FluvioTestRequirements;
+///    let test_reqs : FluvioTestRequirements = [...]
+///    if test_reqs.min_spu > option.spu {
+///        panic!("Test requires {} spu", test_reqs.min_spu)
+///    }
+///     println!("Test")
+/// }
+#[proc_macro_attribute]
+pub fn fluvio_test(args: TokenStream, input: TokenStream) -> TokenStream {
+    let fn_test_reqs: FluvioTestRequirements = parse_macro_input!(args as AttributeArgs).into();
+    let fn_test_reqs_str =
```

Comment:
Idea is that test attribute should be parsed.  With json parsing, this doesn't propagated to parser and we have no feedback

Reply:
See:  https://github.com/infinyon/node-bindgen/blob/master/nj-derive/src/ast/attribute.rs.

Reply:
this looks like syntax problem.  it should been:
```
let output_fn = quote! {

       pub async fn #out_fn_iden(client: Fluvio, option: TestOption) {
                use fluvio_test_util::test_meta::TestRequirements;
        }
};
```

Reply:
It's a syntax problem of from That is the output from cargo. The line numbers jump from line 55 to line 80.

The only difference is the method for passing the `TestRequirements` struct into `quote!{}`

```
let test_reqs : TestRequirements = serde_json::from_str(#fn_test_reqs_str).expect("Could not deserialize test reqs");
```
vs
```
let test_reqs : TestRequirements = #fn_test_reqs
```

Reply:
Define what should be in the generated code then reverse map to template.